flink/v1.19/flink/src/test/java/org/apache/iceberg/flink/sink/TestIcebergFilesCommitter.java

```diff
@@ -887,6 +921,55 @@ public void testCommitTwoCheckpointsInSingleTxn() throws Exception {
     }
   }
 
+  @TestTemplate
+  public void testCommitMultipleCheckpointsWithDuplicateData() throws Exception {
```

Comment:
Nit: maybe a better name, like `testCommitMultipleCheckpointsForV2Table`?
Or anything where we highlight that the goal is to create different commits for different checkpoints, even if the WriteResults arrive at the same time.

Reply:
I'm not satisfied with the name (even the on I suggested), so probably the best would be to add javadoc to the test method describing the scenario we are testing

Reply:
Yes, the triggering conditions are a bit complicated, and it is necessary to add javadoc to describe it clearly.